### PR TITLE
python311Packages.responses: 0.23.3 -> 0.24.0

### PR DESCRIPTION
**Pull request by fabaff**
Diff: https://github.com/getsentry/responses/compare/refs/tags/0.23.3...0.24.0

Changelog: https://github.com/getsentry/responses/blob/0.24.0/CHANGES

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/responses/default.nix [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 buildPythonPackage rec {
   pname = "responses";
-  version = "0.23.3";
+  version = "0.24.0";
   format = "setuptools";
 
   disabled = pythonOlder "3.7";
```
Comment (natsukium) [inline pkgs/development/python-modules/responses/default.nix:23]:
```suggestion
  disabled = pythonOlder "3.8";
```
https://github.com/getsentry/responses/blob/5579f8ecbe4bc59e04bc334f98c8a68b93c57cb3/setup.py#L79

Comment (fabaff) [reply pkgs/development/python-modules/responses/default.nix]: Thanks

**Comment by fabaff** [comment]
@ofborg build python311Packages.responses python311Packages.google-auth python311Packages.detect-secrets
 python311Packages.pygitguardian

File: pkgs/development/python-modules/responses/default.nix [hunk 2]
```diff
@@ -11,24 +11,23 @@
...
 , types-toml
-, typing-extensions
 , urllib3
 }:
 
 buildPythonPackage rec {
   pname = "responses";
-  version = "0.23.3";
+  version = "0.24.0";
   format = "setuptools";
```
Comment (mweinelt) [inline pkgs/development/python-modules/responses/default.nix:20]: Enable the PEP517 build.

**Pull request closed without merging**